sunkit_image/offlimb_enhance.py

```diff
@@ -295,3 +297,162 @@ def normalizing_radial_gradient_filter(
         ) / radial_intensity_distribution_summary[i]
 
     return sunpy.map.Map(data, smap.meta)
+
+
+def fourier_normalizing_radial_gradient_filter(
+    smap,
+    radial_bin_edges,
+    order,
+    attenuation_coefficients,
+    ratio_mix=[15, 1],
+    scale=None,
+    intensity_summary=np.nanmean,
+    intensity_summary_kwargs={},
+    width_function=np.std,
+    width_function_kwargs={},
+    application_radius=1 * u.R_sun,
+    number_angular_segments=130,
+):
+    """
+    Implementation of the fourier normalizing radial gradient filter (FNRGF). 
+
+    ..note ::
+
+        After applying the filter, current plot settings such as the image normalization
+        may have to be changed in order to obtain a good-looking plot
+
+    Parameters
+    ----------
+    smap : `sunpy.map.Map`
+        A SunPy map
+    radial_bin_edges : `~astropy.units.Quantity`
+        A two-dimensional array of bin edges of size [2, nbins] where nbins is
+        the number of bins.
+    order : `~int`
+        Order (Number) of fourier coefficients.
+    attenuation_coefficients : `~float`
+        A two dimensional array of shape [2, order + 1]. The first row contain attenuation coefficients
+        for mean calculations. The second row contains attenuation coefficients for standard deviation
+        calculation.
+    ratio_mix : `~float`
+        The ratio in which the original image and filtered image are mixed.
+    scale : None | `astropy.units.Quantity`
+        The radius of the Sun expressed in map units.  For example, in typical
+        helioprojective Cartesian maps the solar radius is expressed in units
+        of arcseconds.  If None, then the map scale is used.
+    intensity_summary :`~function`
+        A function that returns a summary statistic of the radial intensity,
+        for example `~numpy.mean` and `~numpy.median`.
+    intensity_summary_kwargs : None | `~dict`
```

Comment:
I would check the other docstring again, there are more text and style changes.

Reply:
Okay